### ISSUE TITLE
Unexpected errors in actions

### ISSUE THREAD

**Issue by georgysavva** [golangci/golangci-lint-action#191]
Hi. We are using golangci-lint-action v2.5.1 with golangci-lint v1.38.0
When running golangci-lint in the Github actions it returns weird errors https://github.com/iotaledger/goshimmer/runs/2174488190

```
Running [/home/runner/golangci-lint-1.38.0-linux-amd64/golangci-lint run --out-format=github-actions --timeout=10m] in [] ...
 Error: unlambda: replace `func(c echo.Context) error {
 	return ledgerstateAPI.GetTransaction(c)
 }` with `ledgerstateAPI.GetTransaction` (gocritic)
 Error: unlambda: replace `func(c echo.Context) error {
 	return ledgerstateAPI.GetTransactionMetadata(c)
 }` with `ledgerstateAPI.GetTransactionMetadata` (gocritic)
 Error: unlambda: replace `func(c echo.Context) error {
 	return ledgerstateAPI.GetTransactionAttachments(c)
 }` with `ledgerstateAPI.GetTransactionAttachments` (gocritic)
 Error: unlambda: replace `func(c echo.Context) error {
 	return ledgerstateAPI.GetOutput(c)
 }` with `ledgerstateAPI.GetOutput` (gocritic)
 
 Error: issues found
 Ran golangci-lint in 24151ms
```

The problem is that the code from the alerts doesn't exist in the repository. And when I run golangci-lint locally I don't see any alerts as expected. 
  I tried to disable the cache in actions with `skip-pkg-cache` parameter, but it didn't help I see the same errors in the Github actions log, see https://github.com/iotaledger/goshimmer/runs/2174517258

**Comment by ldez** [comment]
Hello,

I think that your problem is related to that: https://github.com/iotaledger/goshimmer/blob/840aa1213952664684788c8429334805654c40ec/.golangci.yml#L5

as an explicit hash is provided the linter will analyze some dead code.

**Comment by ldez** [comment]
It sounds like you have found the issue https://github.com/iotaledger/goshimmer/pull/1113

**Comment by georgysavva** [comment]
@ldez No, this PR isn't about finding the issue with golangci-lint actions, it's about configuring golangci-lint properly for our project. 
The only workaround for the golangci-lint actions problem that I described above was to use the reviewdog actions: https://github.com/reviewdog/action-golangci-lint

**Comment by NavneethJayendran** [comment]
@georgysavva Just curious if it's possible to workaround by running golangci-lint on your project alone i.e. `./...` or `github.com/iotaledger/goshimmer/...`

**Comment by ldez** [comment]
`skip-pkg-cache` was related to Go cache.

The right option to skip the cache is [`skip-save-cache`](https://github.com/golangci/golangci-lint-action?tab=readme-ov-file#skip-cache).

The cache management has been improved since golangci-lint-action v2.5.1.
